lec15-git-projects.Rmd

```diff
@@ -0,0 +1,415 @@
+> - Collaborating with GitHub (50 min)
+>       - Creating a fork + GitHub setup (5 min)
+>       - Branches and pull requests (40 min)
+>       - Issues (5 min)
+>
+-----
+
+## Intro to Git
+
+Git is a version control system that tracks changes in files. Although
+it is primarily used for software development, Git can be used to track
+changes in any files, and is an indispensable tool for collaborative projects.
+
+Git is often used via [GitHub](http://www.github.com), a website that
+hosts Git repositories. GitHub therefore facilitates collaboration
+by allowing for code review and other useful features. In the spirit of 
+'working open' and ensuring scientific reproducibility, it has also become increasingly 
+common for scientists to upload scripts and related files to GitHub for others to use.
+
+In this lesson, we'll be covering some basic Git commands, before
+moving on to using GitHub for the upcoming group projects.
+The members of group will be given admin access to a pre-made repository. 
+
+### Command line Git
+
+#### Setup and Installation
+
+Git is primarily used through a _bash interpreter_, also known as a 'bash shell', 
+'the command line', or 'the terminal'. While bash interpreters are 
+natively available on Mac and Linux operating systems, Windows users may 
+have to externally install a bash shell.
+
+To install Git on a **Mac**, you will have to install [Xcode](https://itunes.apple.com/ca/app/xcode/id497799835?mt=12),
+which can be downloaded from the Mac App store. Be warned: Xcode is a very 
+large download (approximately 6 GB)! Install Xcode and Git should be
+ready for use.
+
+To install Git (and Bash) on **Windows**, download [Git for Windows](https://gitforwindows.org/) from its [website](https://github.com/git-for-windows/git/releases/tag/v2.19.1.windows.1). 
+At the time of writing, 2.19.1 is the most recent version. Download the `exe`
+file to get started. Git for Windows provides a program called Git Bash, which
+provides a bash interpreter that can also run Git commands. 
+
+To install Git for **Linux**, use the preferred package manager
+for your distribution and follow the instructions [listed here](https://git-scm.com/download/linux). 
+
+To test whether Git has successfully been installed, open a bash interpreter, type
+
+```
+git --version
+```
+
+and hit enter. If the interpreter returns a version number, Git has been
+installed.
+
+#### Bash Primer
+
+Before we can use Git in the bash interpreter, it's worth learning a
+few basic bash commands. All bash commands are run simply by typing them 
+into the interpreter and pressing enter. 
+
+```
+pwd
+```
+
+`pwd`, or 'print working directory', will return the folder bash
+is currently in. This is similar to the concept of the working directory in R
+that we discussed earlier in the course.
+
+```
+cd [folder]
+```
+
+`cd` changes the current working directory. This is useful in tandem with
+`~`, which is shorthand for 'home directory' in bash (usually 
```

Comment:
Does `cd ~` work on Windows as well? According to [this](https://stackoverflow.com/questions/9228950/what-is-the-alternative-for-users-home-directory-on-windows-command-prompt), the Windows equivalent of `~` is `%userprofile%`. Hopefully `~` works in git bash because otherwise that would be super annoying.

Reply:
I'm not 100% sure, but no one using Git Bash on the lab computers last Thursday reported an issue with commands that included `~`, so I think we should be good on this front

Reply:
I think we've always been fine at SWC? This is Git-Bash, not a Command Prompt, after all.